src/nanoarrow/common/array.c

```diff
@@ -642,6 +673,10 @@ void ArrowArrayViewReset(struct ArrowArrayView* array_view) {
     ArrowFree(array_view->union_type_id_map);
   }
 
+  if (array_view->variadic_buffer_views != NULL) {
+    ArrowFree(array_view->variadic_buffer_views);
```

Comment:
Assuming we go down the path of `ArrowArrayViewBufferView(array_view, i)` do you think its worth moving existing code away from using the `buffer_views` member? I think part of the awkwardness right now is how heavily that is used, but since it is a stack allocated array of `BufferView` objects it can't be re-used here